doc/architecture/configuration-and-management.rst

```diff
@@ -0,0 +1,208 @@
+actual files, and because the Configuration Fragments are stored in JSON or
+XML, the CRUD operations are trivial.
+
+The Installer
+-------------
+
+The task of the Installer is to generate the actual configuration
+files, typically under ``/etc``, for the various Ostro OS software
+components. The Installer tracks configuration data changes in the
+Drop Zone using ``inotify``. When a Configuration
+Fragment changes, the Installer reads the altered configuration
+values and generates new versions of the affected configuration files
+incorporating the new values. The configuration file generation is
+template based.
+
+A single value change in a Configuration Fragement might lead to the
+re-generation of multiple configuration files of various Ostro OS
+components. For instance, if the listening port of the SSH Daemon is
+changed, the new value should be incorporated into the
+``/lib/systemd/system/sshd.socket`` and ``/etc/sshd_config`` files, and
+the Firewall rules should be modified too.
+
+Each configuration file, that depends on configuration value changes,
+needs to have a dedicated *Template* and a so called *Resource
+Description* file. The Template is used to generate the configuration
+file while the Resource Description specifies the context for the
+operation:
+
+-  the resulting configuration file (the output)
+-  the template used to regenerate the output
+-  regeneration triggers (a list of Configuration Fragments that
+   trigger re-generation)
+-  an optional executable program that can validate the generated configuration
+   file
+-  an optional executable program that can install the new configuration version, 
+   for example, ``systemctl reload xxxx.service``
+
+Data Model
+==========
+
+The tree of Configuration Fragments is composed of three subtrees
+with an identical structure: Factory, Common,  and Local.
+
+* The Factory subtree contains the default values of the Ostro OS
+  distribution. This subtree is part of the OS image and is
+  immutable.
+
+* The Common subtree hosts the values that are common for a larger set
+  of systems. For instance, a service provider who maintains a large
+  number of Ostro OS based systems could store configuration values under
+  the Common subtree that are identical (and hence common) for all systems.
+
+* The Local subtree meant for individual Ostro OS device-specific
+  configuration values. Configuration values under the Local subtree can
+  be set either remotely by Service Providers or locally by end-users
+  using their phones, tablets, or laptops. (WLAN SSID is an example
+  of locally set configuration data.)
+
+Templates offer mechanisms to deep-merge these subtrees in increasing
+priority order of Factory, then Common, and finally Local. The deep-merge
+means that the merging is done on individual configuration value level as
+opposed to Configuration Fragment level.
+
+Components
+==========
+
+The Ostro OS Configuration & Management system is an optional yocto layer 

```

Comment:
The Yocto Project team wants all references to "yocto" to be "Yocto Project", so this should say "Yocto Project layer"